server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -389,6 +390,19 @@ public boolean isDataStreamTimestampFieldEnabled() {
         return dtfm != null && dtfm.isEnabled();
     }
 
+    /**
+     * Returns if this mapping contains a timestamp field that is of type date and indexed
+     * @return {@code true} if contains an indexed timestamp field of type date, {@code false} otherwise.
+     */
+    public boolean hasTimestampField() {
+        final MappedFieldType mappedFieldType = fieldTypesLookup().get(DataStream.TimestampField.FIXED_TIMESTAMP_FIELD);
+        if (mappedFieldType instanceof DateFieldMapper.DateFieldType) {
+            return mappedFieldType.isSearchable();
```

Comment:
Should we also check if doc values are enabled ?

Reply:
Addressed in f852ec983a3b3dc670b308f6215f612424b27e21